src/peft/tuners/xlora/model.py

```diff
@@ -86,6 +87,70 @@ def convert_layers_to_xlora(
     return (total_swapped, device)
 
 
+def _load_adapter_into_lora_model(
+    lora_model: LoraModel,
+    adapter_name: str,
+    model_id: str,
+    i: int,
+    torch_device: Optional[str] = None,
+    ephemeral_gpu_offload: bool = False,
+    autocast_adapter_dtype: bool = True,
+    subfolder: Optional[str] = None,
+    **kwargs,
+):
+    """
+    All params pertain to the adapter (adapter name, model id, `i` is the adapter number in 0 indexing)
+    """
+    from peft.peft_model import PeftModel
```

Comment:
Let's add a comment here that this "emulates" the functionality of `PeftModel.from_pretrained` and might need to be updated if there are changes there.

Reply:
Thanks, added that.